### PR TITLE
Log more context to segment

### PR DESCRIPTION
**Pull request by gregdingle**
# Description

This makes full use of segment in the client and server given the data we have at call-time. 

# Test

Load a page and see log data such as 
```
{
  "email": "gdingle@chanzuckerberg.com",
  "name": "Greg Dingle",
  "created_at": "2018-06-08T13:20:09.000-07:00",
  "updated_at": "2019-02-21T11:44:53.000-08:00",
  "role": 1,
  "allowed_features": [
    "data_discovery"
  ],
  "institution": null,
  "admin": true,
  "demo_user": false,
  "biohub_user": false,
  "czi_user": true,
  "createdAt": "2018-06-08T13:20:09-07:00",
  "firstName": "Greg",
  "lastName": "Dingle",
  "sign_in_count": 3,
  "current_sign_in_at": "2019-02-21T11:44:53.000-08:00",
  "last_sign_in_at": "2019-02-08T15:08:14.000-08:00",
  "current_sign_in_ip": "172.19.0.1",
  "last_sign_in_ip": "172.19.0.1"
}
```

### CODE REVIEW

File: app/lib/metric_util.rb [hunk 1]
```diff
@@ -24,76 +27,116 @@ class MetricUtil
...
         user,
-        count: entries.count, project_id: project_id, host_genome_id: host_genome_id, client: client
+        { count: entries.count, project_id: project_id, host_genome_id: host_genome_id, client: client },
+        request
       )
     end
   end
+
+  # Start private class methods
+  class << self
```
Comment (tfrcarvalho) [inline app/lib/metric_util.rb]: Why do you need this versus just using the private keyword?

File: app/models/user.rb [hunk 2]
```diff
@@ -69,11 +69,62 @@ def can_upload(s3_path)
...
-      unless ["czbiohub.org", "ucsf.edu"].include?(email.split("@").last)
+      unless biohub_user?
         return false
       end
     end
 
     true
   end
+
+  def biohub_user?
```
Comment (jshoe) [inline app/models/user.rb]: The string checking was a little hacky to begin with, so in the future we could lean on 'institution' and backfill it for Biohub/UCSF/CZI

File: app/views/layouts/_page_header.html.erb [hunk 3]
```diff
@@ -11,9 +11,12 @@
         userSignedIn: true
       }, 'page_header', JSON.parse('<%= raw escape_json(request_context)%>'))
 
+
       <% if ENV["SEGMENT_JS_ID"] %>
         <!-- Associate user ID with Segment -->
-        analytics.identify(<%= current_user.id %>);
+        analytics.identify(
+          <%= current_user.id %>,
+          <%= raw escape_json(current_user.traits_for_segment) %>);
```
Comment (jshoe) [inline app/views/layouts/_page_header.html.erb]: out of curiosity do you see this triggering another db request for every request in your local logs?

Comment (gregdingle) [reply app/views/layouts/_page_header.html.erb]: Yes, that's why I did https://github.com/chanzuckerberg/idseq-web/pull/2088 .

File: app/models/user.rb (hunk 2 above)
Comment (gregdingle) [reply app/models/user.rb]: Yes, institution seems valuable in the long run. Is it currently required and standardized?

Comment (jshoe) [reply app/models/user.rb]: it is required on the "interest form" but account creation either happens through 1) Product creating the user at /users/new or 2) many accounts created through the Add User To Project feature which auto-creates and invites users. I think we could update both forms to require institution

**Pull request merged**